pkg/graveler/graveler.go

```diff
@@ -28,6 +30,37 @@ const (
 	DiffTypeConflict
 )
 
+type RefModType rune
+
+const (
+	RefModTypeTilde  RefModType = '~'
+	RefModTypeCaret  RefModType = '^'
+	RefModTypeAt     RefModType = '@'
+	RefModTypeDollar RefModType = '$'
+)
+
+type RefModifier struct {
+	Type  RefModType
+	Value int
+}
+
+type ParsedRef struct {
+	BaseRef      string
+	RefModifiers []RefModifier
```

Comment:
The order is important? It's worth mentioning it if that's the case.

Reply:
will add it part of the struct comment